File: backend/src/main/java/com/davnig/invoicemanagementapi/config/BeansConfig.java
package com.davnig.invoicemanagementapi.config;

import com.querydsl.jpa.impl.JPAQueryFactory;
import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class BeansConfig {

    @PersistenceContext
    private EntityManager entityManager;

    @Bean
    public JPAQueryFactory getJPAQueryFactory() {
        return new JPAQueryFactory(entityManager);
    }

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/config/BeansConfig.kt
package com.davnig.invoicemanagementapi.config

import com.querydsl.jpa.impl.JPAQueryFactory
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class BeansConfig {
    @PersistenceContext
    private val entityManager: EntityManager? = null

    @get:Bean
    val jPAQueryFactory: JPAQueryFactory
        get() = JPAQueryFactory(entityManager)
}